plugins/aea-cli-benchmark/aea_cli_benchmark/case_acn_communication/case.py

```diff
@@ -0,0 +1,208 @@
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Check amount of time for acn connection communications."""
+import asyncio
+import logging
+import os
+import time
+from contextlib import contextmanager
+from tempfile import TemporaryDirectory
+from typing import Callable, List, Tuple, Union
+
+from aea_cli_benchmark.case_acn_communication.utils import (
+    DEFAULT_DELEGATE_PORT,
+    DEFAULT_MAILBOX_PORT,
+    DEFAULT_NODE_PORT,
+    _make_libp2p_client_connection,
+    _make_libp2p_connection,
+    _make_libp2p_mailbox_connection,
+)
+
+from aea.connections.base import Connection
+from aea.mail.base import Envelope
+
+from packages.fetchai.protocols.default.message import DefaultMessage
+
+
+class TimeMeasure:
+    """Time measure data class."""
+
+    def __init__(self):
+        """Init data class instance."""
+        self.time = -1
+
+
+@contextmanager
+def time_measure():
+    """Get time measure context."""
+    start = time.time()
+    m = TimeMeasure()
+    try:
+        yield m
+    finally:
+        m.time = time.time() - start
+
+
+def make_envelope(from_addr: str, to_addr: str) -> Envelope:
+    """Construct an envelope."""
+    msg = DefaultMessage(
+        dialogue_reference=("", ""),
+        message_id=1,
+        target=0,
+        performative=DefaultMessage.Performative.BYTES,
+        content=b"hello",
+    )
+    envelope = Envelope(to=to_addr, sender=from_addr, message=msg,)
+    return envelope
+
+
+async def _run(con_maker: Callable[..., Connection]) -> Tuple[float, float]:
+    """Run test case and return times for the first and the second messages sent over ACN."""
+    try:
+        connections = []
+        genesis_node = _make_libp2p_connection(".", relay=True)
+        await genesis_node.connect()
+        connections.append(genesis_node)
+        genesis_multiaddr = genesis_node.node.multiaddrs[0]
+
+        relay_node1 = _make_libp2p_connection(
+            ".",
+            relay=True,
+            entry_peers=[genesis_multiaddr],
+            port=DEFAULT_NODE_PORT + 1,
+            mailbox=True,
+            delegate=True,
+            mailbox_port=DEFAULT_MAILBOX_PORT + 1,
+            delegate_port=DEFAULT_DELEGATE_PORT + 1,
+        )
+        await relay_node1.connect()
+        connections.append(relay_node1)
+        relay_node2 = _make_libp2p_connection(
+            ".",
+            relay=True,
+            entry_peers=[genesis_multiaddr],
+            port=DEFAULT_NODE_PORT + 2,
+            mailbox=True,
+            delegate=True,
+            mailbox_port=DEFAULT_MAILBOX_PORT + 2,
+            delegate_port=DEFAULT_DELEGATE_PORT + 2,
+        )
+        await relay_node2.connect()
+        connections.append(relay_node2)
+
+        relay_node1_multiaddr = relay_node1.node.multiaddrs[0]
+        relay_node2_multiaddr = relay_node2.node.multiaddrs[0]
+        await asyncio.sleep(1)
```

Comment:
Wouldn't this mess up the time accuracy?

Reply:
no, time counting is done below